torch/_refs/__init__.py

```diff
@@ -2454,6 +2454,20 @@ def as_strided(
     return prims.as_strided(a, size, stride, storage_offset)
 
 
+@register_decomposition(torch.ops.aten.as_strided_scatter)
+def as_strided_scatter(
+    input: TensorLikeType,
+    src: TensorLikeType,
+    size: ShapeType,
+    stride: StrideType,
+    storage_offset: Optional[int] = None,
+) -> TensorLikeType:
+    output = input.clone()
+    output_view = output.as_strided(size, stride, storage_offset)
+    output_view.copy_(src)
```

Comment:
Is it fine to use `copy_`, or should we use `copy_to`? @mruberry

Reply:
This is a really interesting implementation. Why is this not `as_strided` followed by `clone`?

Reply:
That is, it seems like there should be an out-of-place way to write this in terms of primitive operations, and if there's not that suggest we're missing a prim -- what prim are we missing in that case?

Reply:
> Why is this not `as_strided` followed by clone? 

Because the values in `input` may end up in the output if the view skips elements. For example,
```
import torch
a = torch.arange(6)
b = torch.zeros(3)
c = torch.as_strided_scatter(a, b, size=b.shape, stride=(2,))
print(c) # tensor([0, 1, 0, 3, 0, 5])
```

Reply:
By the way, this operator is a primitive used by the functionalization pass to map in-place operations on views into their functional equivalent.

Reply:
Should we make a primitive `as_strided_scatter`, then?